lib/core/test/unit/Cardano/WalletSpec.hs

```diff
@@ -402,10 +403,10 @@ walletGetProp newWallet = monadicIO $ liftIO $ do
 walletGetWrongIdProp
     :: ((WalletId, WalletName, DummyState), WalletId)
     -> Property
-walletGetWrongIdProp (newWallet, corruptedWalletId) = monadicIO $ liftIO $ do
+walletGetWrongIdProp (newWallet@(wid, _, _), walletId) = monadicIO $ liftIO $ do
     (WalletLayerFixture _db wl _walletIds _) <- liftIO $ setupFixture newWallet
-    attempt <- runExceptT $ W.readWallet wl corruptedWalletId
-    attempt `shouldSatisfy` isLeft
+    attempt <- runExceptT $ W.readWallet wl walletId
+    attempt `shouldSatisfy` if wid /== walletId then isLeft else isRight
```

Comment:
I'm guessing this won't work, as `if` is expecting a `Bool` here, but the result of evaluating `=/=` is a `Property`.

Perhaps:
```suggestion
    attempt `shouldSatisfy` if wid /= walletId then isLeft else isRight
```

Reply:
oh yes, start compile & test, submit PR, then forget! fixed now.